crates/sui-json-rpc/src/traffic_control.rs

```diff
@@ -42,15 +48,31 @@ where
     fn call(&self, req: jsonrpsee::types::Request<'a>) -> Self::Future {
         let service = self.inner.clone();
         let traffic_controller = self.traffic_controller.clone();
-
+        let policy_config = self.policy_config.clone();
         async move {
             if let Some(traffic_controller) = traffic_controller {
                 let client = req.extensions().get::<IpAddr>().cloned();
                 if let Err(response) = handle_traffic_req(&traffic_controller, &client).await {
                     response
                 } else {
+                    // Scale the weight based on the request size
+                    let spam_weight = if policy_config.unwrap().weight_spam_by_request_size {
+                        println!("TESTING -- weighting spam by request size");
+                        let params_count = req.params.as_ref().and_then(|p| {
+                            let raw_value = p.get();
+                            let params: Option<Vec<serde_json::Value>> =
+                                serde_json::from_str(raw_value).ok();
```

Comment:
this won't handle nested params, right? I no it is a bit more fragile and has a lot of boilerplate but i wonder if the request handler needs to explicitly determine a param count?